tests/touch/touch.ino

```diff
@@ -0,0 +1,196 @@
+#include <unity.h>
+#include "soc/soc_caps.h"
+
+#if SOC_TOUCH_SENSOR_NUM > 0
+
+#include "driver/touch_pad.h"
+
+#if CONFIG_IDF_TARGET_ESP32
+
+#define TEST_TOUCH_CHANNEL   (9)
+static touch_pad_t touch_list[TEST_TOUCH_CHANNEL] = {
+    TOUCH_PAD_NUM0,
+    //TOUCH_PAD_NUM1 is GPIO0, for download.
+    TOUCH_PAD_NUM2,
+    TOUCH_PAD_NUM3,
+    TOUCH_PAD_NUM4,
+    TOUCH_PAD_NUM5,
+    TOUCH_PAD_NUM6,
+    TOUCH_PAD_NUM7,
+    TOUCH_PAD_NUM8,
+    TOUCH_PAD_NUM9
+};
+
+uint8_t TOUCH_GPIOS[] = {4,2,15,13,12,14,27,33,32};
+
+#define NO_TOUCH_GPIO 25
+
+#define RELEASED_VALUE        80 //80+ read value to pass test
+#define PRESSED_VALUE         20 //20- read value to pass test
+#define INTERRUPT_THRESHOLD   40
+
+#else //ESP32S2 and ESP32S3
+
+#define TEST_TOUCH_CHANNEL   (12) //14
+static touch_pad_t touch_list[TEST_TOUCH_CHANNEL] = {
+    TOUCH_PAD_NUM1,
+    TOUCH_PAD_NUM2,
+    TOUCH_PAD_NUM3,
+    TOUCH_PAD_NUM4,
+    TOUCH_PAD_NUM5,
+    TOUCH_PAD_NUM6,
+    TOUCH_PAD_NUM7,
+    TOUCH_PAD_NUM8,
+    TOUCH_PAD_NUM9,
+    TOUCH_PAD_NUM10,
+    TOUCH_PAD_NUM11,
+    TOUCH_PAD_NUM12
+    //TOUCH_PAD_NUM13, //Wrong reading
+    //TOUCH_PAD_NUM14
+};
+
+uint8_t TOUCH_GPIOS[] = {1,2,3,4,5,6,7,8,9,10,11,12/*,13,14*/};
+
+#define NO_TOUCH_GPIO 17
+
+#if CONFIG_IDF_TARGET_ESP32S2
+  #define RELEASED_VALUE        10000 //10000- read value to pass test
+  #define PRESSED_VALUE         42000 //40000+ read value to pass test
+  #define INTERRUPT_THRESHOLD   30000
+#elif CONFIG_IDF_TARGET_ESP32S3
+  #define RELEASED_VALUE        25000 //25000- read value to pass test
+  #define PRESSED_VALUE         100000 //150000+ read value to pass test
+  #define INTERRUPT_THRESHOLD   80000
+#endif
```

Comment:
I would suggest to add another `#else` here and an error, in order to catch future chips with touch functionality.

Reply:
@me-no-dev Did you mean something like this? So the test will not compile or fail for new chip (so we now it needs to be adjusted)? All tests will fail with these values (chip is using TOUCH_V2).
```
#else //defaults for upcoming chips (needs to be adjusted)
  #define RELEASED_VALUE        -1     
  #define PRESSED_VALUE         0xFFFFFFFF 
  #define INTERRUPT_THRESHOLD   -1      
#endif
```

Reply:
could even be as easy as 
```cpp
#else
  #error Test not currently supported on this chip. Please adjust and try again
#endif
```

Reply:
Done :) best and easiest option. Thanks